datalake/src/main/java/com/sequenceiq/datalake/flow/SdxReactorFlowManager.java

```diff
@@ -116,6 +119,41 @@ public FlowIdentifier triggerDatalakeRuntimeUpgradeFlow(SdxCluster cluster, Stri
         }
     }
 
+    /**
+     * Checks if Sdx backup can be performed.
+     * Uses Cloud storage file system type to find the cloud provider.
+     * @param cluster Sdx cluster
+     * @return true if backup can performed, False otherwise.
+     */
+    private boolean shouldSdxBackupBePerformed(SdxCluster cluster) {
+        boolean retVal = true;
+        String reason = null;
+        if (cluster.isRangerRazEnabled()) {
+            retVal = false;
+            reason = "RAZ is enabled";
+        } else if (isVersionOlderThan(cluster, "7.2.1")) {
+            retVal = false;
+            reason = "Unsupported runtime: " + cluster.getRuntime();
+        } else if (cluster.getCloudStorageFileSystemType().isGcs()) {
+            retVal = false;
+            reason = "Unsupported cloud provider GCS ";
+        } else if (cluster.getCloudStorageFileSystemType().isAdlsGen2() &&
+                isVersionOlderThan(cluster, "7.2.2")) {
+            retVal = false;
+            reason = "Unsupported cloud provider Azure on runtime: " + cluster.getRuntime();
+        }
+        if (!retVal) {
+            LOGGER.info("Backup not triggered. Reason: " + reason);
+        }
+        return retVal;
+}
+
+    private static boolean isVersionOlderThan(SdxCluster cluster, String baseVersion) {
```

Comment:
pls use CMRepositoryVersionUtil#isVersionOlderThanLimited() instead

Reply:
Both of these using the same versionComparator. I initially started using the same API. I had to make the validation login complex if i had to use CMRepositoryVersionUtil that is why i deliberately not use it.